lib/analysis/interproc-analysis/SVFUtils.h

```diff
@@ -0,0 +1,159 @@
+//
+// Created by Kostas Ferles on 10/6/22.
+//
+
+#ifndef VANGUARD_SVFUTILS_H
+#define VANGUARD_SVFUTILS_H
+
+#include <map>
+#include <set>
+
+#include <SVF-FE/LLVMUtil.h>
+#include <Util/Options.h>
+#include <Graphs/SVFG.h>
+#include <WPA/Andersen.h>
+#include <SVF-FE/SVFIRBuilder.h>
+#include <Graphs/SVFGNode.h>
+
+namespace analysis {
+
+    class SVFUtils {
+    public:
+
+        static bool dependsOnTrg(const llvm::Value *src, const llvm::Value *trg) {
+            if (cachedDeps.find(src) == cachedDeps.end()) {
+                std::set<const SVF::VFGNode*> deps;
+                return traverseMemDependencies(src, deps, trg);
+            }
+            else {
+                auto deps = cachedDeps[src];
```

Comment:
nit: I'd make this explicitly a reference
```suggestion
                auto &deps = cachedDeps[src];
```